Add JsonSchema support

Ronserruya opened the pull request:
Following up on #5 
I added support for JsonSchema, 

- [x] Unittests pass
- [x] flake and mypy linting and type check pass
- [x] Kafka integration tests pass
- [x] Readme updated to include JSON schema support
- [x] Bumped version to 1.1.0
- [x] Java compatibility tests
 
~I wasn't able to test the java compatibility test. (Had some issues with the compilation, and either way, I wouldn't really know what to test since I'm not familiar with the java library)~

Please let me know if there is anything else that needs fixing.
Thanks again :) 👍

cdebost commented on tests/integration/kafka/test_kafka_integration.py:
```diff
@@ -65,12 +69,25 @@ def test_produce_consume_with_ser_de_schema_registry(
     )
+
+    # jsonschema has no fqn, so we use topic_name_strategy for it
+    # (which also requires a separate producer)
```
Interesting. I'd assumed jsonschema had some kind of record name property like Avro does. I suppose this means the naming strategies are actually an Avro-specific feature, not something applicable to the whole serializer. Not sure how to change the API so it's still straightforward to set the naming strategy for Avro, without being misled into thinking the same strategies can apply to json too. Maybe just prefix the property with `avro_` will do...

Could you describe what your use case is at a high level? I'm guessing you don't have a requirement to mix incompatible schemas on the same topic?

Ronserruya replied:
In a high level we plan to send a json/avro(not yet decided) msg to a service using Kafka, that service handles that message and writes another message using kafka, which another service reads, and so on in a potential pipeline of services.

We will not mix different schema data types in the same topic.
All of the schema operations (create/delete/retrieve) will probably happen using different methods (due to permissions and other stuff), so I think we will only use the JsonSchema/AvroSchema classes and the encode/decode method to stay compatible with msg schema used by this/java pkg (first magic bit, then compression bit, than schema...)

Regarding the naming API, since you get the schema in the method I guess you can raise an error in case the strategy requires a name, or add an "avro_naming_strategy" and "json_naming_strategy" arguments to the serializer

cdebost replied:
I suppose that some users will add an extra field to their json schemas that indicates the name of the schema, and they'd want the deserializer to use that. That means the user needs a way to customize how `JsonSchema`s are constructed, probably by telling the serializer how to pick a FQN for a new schema. The deserializer should also set the FQN appropriately on downloaded schemas, which is more tricky because it has to figure out which strategy was used to name the schema and whether it is able to mine the FQN out of that (or not).

I think this PR is fine as is. I don't have a requirement for naming strategies with json schema and it's going to require a bit more thought. In the short term the naming strategies ought to throw an error or at least log a warning if they are given a schema that doesn't have a FQN.

cdebost reviewed (commented):
Thanks @Ronserruya.

Just a question re: schema naming strategies. Also, could you explain what compilation error you had with the Java compatibility test? I tried to make it mostly autonomous assuming java/maven is installed, but you're the first other person that is going through the process.

Ronserruya commented:
> Thanks @Ronserruya.
> 
> Just a question re: schema naming strategies. Also, could you explain what compilation error you had with the Java compatibility test? I tried to make it mostly autonomous assuming java/maven is installed, but you're the first other person that is going through the process.

Solved it, I just had an issue with the java version (8 instead of 11), and even after installing the script used the wrong version when running, so I just used the full path of my installation. I guess you can set up a docker with everything needed for this test if you want other people to avoid these conflicts.

Anyway, the tests now work, and I added a test that uses the JSON support of the java pkg as well.
In addition, while doing that, I noticed that the java pkg validates the JSON payload on every read and write, so I added this feature as well

cdebost reviewed (approved):
LGTM

cdebost commented:
Closes #5.

cdebost commented:
Released at https://pypi.org/project/aws-glue-schema-registry/1.1.0/
